crypto/fipsmodule/rand/ctrdrbg.c

```diff
@@ -63,16 +249,69 @@ int CTR_DRBG_init(CTR_DRBG_STATE *drbg,
   }
 
   // Section 10.2.1.2
+  for (size_t i = 0; i < sizeof(kInitMask); i++) {
+    seed_material[i] ^= kInitMask[i];
+  }
 
-  // kInitMask is the result of encrypting blocks with big-endian value 1, 2
-  // and 3 with the all-zero AES-256 key.
-  static const uint8_t kInitMask[CTR_DRBG_ENTROPY_LEN] = {
-      0x53, 0x0f, 0x8a, 0xfb, 0xc7, 0x45, 0x36, 0xb9, 0xa9, 0x63, 0xb4, 0xf1,
-      0xc4, 0xcb, 0x73, 0x8b, 0xce, 0xa7, 0x40, 0x3d, 0x4d, 0x60, 0x6b, 0x6e,
-      0x07, 0x4e, 0xc5, 0xd3, 0xba, 0xf3, 0x9d, 0x18, 0x72, 0x60, 0x03, 0xca,
-      0x37, 0xa6, 0x2a, 0x74, 0xd1, 0xa2, 0xf5, 0x8e, 0x75, 0x06, 0x35, 0x8e,
-  };
+  drbg->ctr = aes_ctr_set_key(&drbg->ks, NULL, &drbg->block, seed_material, 32);
+  OPENSSL_memcpy(drbg->counter, seed_material + 32, 16);
+  drbg->reseed_counter = 1;
+
+  return 1;
+}
+
+static void nonce_increment(uint8_t counter[CTR_DRBG_NONCE_LEN]) {
+  size_t index = CTR_DRBG_NONCE_LEN, carry = 1;
```

Comment:
why is `carry a size_t`? `counter` is 16 bytes long and you're casting `carry` to a single byte, lopping off the 48 high bits (assuming 64 bit system)